cdap-distributions/src/hdinsight/pageblob-configure.sh

```diff
@@ -0,0 +1,142 @@
+    if [ -n ${currentValue} ]; then
+      if echo ${currentValue} | grep -q '/cdap'; then
+        echo "fs.azure.page.blob.dir already contains /cdap"
+        return 0
+      else
+        newValue="${currentValue},/cdap"
+      fi
+    else
+      newValue="/cdap"
+    fi
+    echo "Updating fs.azure.page.blob.dir to ${newValue}"
+    setAmbariConfig 'core-site' 'fs.azure.page.blob.dir' ${newValue} || die "Could not update Ambari config"
+    restartCdapDependentClusterServices
+}
+
+# Stop an Ambari cluster service
+stopServiceViaRest() {
+    if [ -z "${1}" ]; then
+        echo "Need service name to stop service"
+        exit 136
+    fi
+    SERVICENAME=${1}
+    echo "Stopping ${SERVICENAME}"
+    curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "INSTALLED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME}
+}
+
+# Start an Ambari cluster service
+startServiceViaRest() {
+    if [ -z "${1}" ]; then
+        echo "Need service name to start service"
+        exit 136
+    fi
+    sleep 2
+    SERVICENAME=${1}
+    echo "Starting ${SERVICENAME}"
+    startResult=$(curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "STARTED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME})
+    if [[ ${startResult} == *"500 Server Error"* || ${startResult} == *"internal system exception occurred"* ]]; then
+        sleep 60
+        echo "Retry starting ${SERVICENAME}"
+        startResult=$(curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "STARTED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME})
+    fi
+    echo ${startResult}
+}
+
+# Restart Ambari cluster services
+restartCdapDependentClusterServices() {
+    stopServiceViaRest HBASE
+    stopServiceViaRest YARN
+    stopServiceViaRest MAPREDUCE2
+    stopServiceViaRest HDFS
+
+    startServiceViaRest HDFS
+    startServiceViaRest MAPREDUCE2
+    startServiceViaRest YARN
+    startServiceViaRest HBASE
+}
+
+
+# Begin Ambari Cluster Prep
+
+USERID=$(echo -e "import hdinsight_common.Constants as Constants\nprint Constants.AMBARI_WATCHDOG_USERNAME" | python)
+echo "USERID=$USERID"
+
+PASSWD=$(echo -e "import hdinsight_common.ClusterManifestParser as ClusterManifestParser\nimport hdinsight_common.Constants as Constants\nimport base64\nbase64pwd = ClusterManifestParser.parse_local_manifest().ambari_users.usersmap[Constants.AMBARI_WATCHDOG_USERNAME].password\nprint base64.b64decode(base64pwd)" | python)
```

Comment:
Move these constants up to the top of the file and use `python -c` (replace `\n` with `;`) rather than echo